src/transformers/models/auto/modeling_auto.py

```diff
@@ -382,6 +383,7 @@
     [
         # Model for Causal LM mapping
         ("bart", "BartForCausalLM"),
+        ("beit3", "Beit3ForCaptioning"),
```

Comment:
this does not seem correct. `Beit3ForCaptioning` has its own architecture and it not a causal LM

Reply:
No, It does not have generate, I am just realising this. In general should have some check somewhere to validate this ?